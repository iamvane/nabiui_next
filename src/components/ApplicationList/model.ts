import { AvailabilityType } from '../Availability/model';

export interface Application {
  applicationId: number;
  instructorId: number;
  applicationMessage: string;
  applicationRate: string;
  availability: AvailabilityType;
  avatar: string;
  displayName: string;
  reviews: number;
  yearsOfExperience: number;
  age: number;
  backgroundCheckStatus: string;

}
export interface ApplicationListType {
  id: number;
  requestTitle: string;
  dateCreated: string;
  applications: Application[];
  freeTrial: boolean;
}
