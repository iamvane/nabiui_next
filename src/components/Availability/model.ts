export interface Availability {
  day: string;
  from: string;
  to: string;
}

export interface AvailabilityType {
  mon8to10?: boolean;
  mon10to12?: boolean;
  mon12to3?: boolean;
  mon3to6?: boolean;
  mon6to9?: boolean;
  tue8to10?: boolean;
  tue10to12?: boolean;
  tue12to3?: boolean;
  tue3to6?: boolean;
  tue6to9?: boolean;
  wed8to10?: boolean;
  wed10to12?: boolean;
  wed12to3?: boolean;
  wed3to6?: boolean;
  wed6to9?: boolean;
  thu8to10?: boolean;
  thu10to12?: boolean;
  thu12to3?: boolean;
  thu3to6?: boolean;
  thu6to9?: boolean;
  fri8to10?: boolean;
  fri10to12?: boolean;
  fri12to3?: boolean;
  fri3to6?: boolean;
  fri6to9?: boolean;
  sat8to10?: boolean;
  sat10to12?: boolean;
  sat12to3?: boolean;
  sat3to6?: boolean;
  sat6to9?: boolean;
  sun8to10?: boolean;
  sun10to12?: boolean;
  sun12to3?: boolean;
  sun3to6?: boolean;
  sun6to9?: boolean;
  [key: string]: any;
}