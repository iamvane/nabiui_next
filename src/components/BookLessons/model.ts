export interface BookLessonPackages {
  name: string;
	lessonNumber: number;
	value: string;
}

export interface BookLessonsPayload {
	studentId: number;
	package: string;
	paymentMethodId?: number;
	paymentMethodCode?: string;
}

export interface BookLessonsData {
	clientSecret?: string;
	lessonRate?: number
	lessonsPrice?: number;
	paymentMethods?: any[]
	placementFee?: number;
	processingFee?: number;
	subTotal?: number
	total?: number;
	freeTrial?: boolean;
	virtuosoDiscount?: number;
	discounts?: number;
	instructor: {
		avatar: string;
		reviews: {
			rate: string;
			quantity: number
		},
		backgroundCheckStatus: string;
		display_name: string;
		rate: number;
		yearsOfExperience: number;
		age: number;
	}
}

export enum LessonStatusType {
	Scheduled = 'scheduled',
	Missed = 'missed',
	Conplete = 'complete'
}

export interface LessonType {
	studentId?: number;
	date?: string;
	time?: string;
	status?: LessonStatusType;
	grade?: number;
	comment?: string;
}
