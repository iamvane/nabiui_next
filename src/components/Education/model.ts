
export interface EducationType {
  id?: number;
  school: string;
  graduationYear: string;
  degreeType: string;
  fieldOfStudy: string;
  schoolLocation: string;
}
