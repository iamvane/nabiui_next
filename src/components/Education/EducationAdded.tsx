import * as React from 'react';

import {
  Icon,
  IconButton,
  Grid,
  Typography
} from '@material-ui/core';

import { EducationType }  from 'components/Education/model';
import {
  degreeTypeLabels,
  EducationAddedComponent
} from 'components/Education/constants';

interface Props extends EducationType {
  deleteEducation: (educationId: number | undefined) => void;
  editEducation: (educationId: number | undefined) => void;
  gridWidth?: boolean | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | undefined;
  notEditable?: boolean;
}

const EducationAdded: React.StatelessComponent<Props> = props => {
  return (
    <Grid
      item={true}
      md={props.gridWidth && props.gridWidth}
      xs={12}
      className={`${props.notEditable ? '' : 'nabi-editable-item'} nabi-margin-top-small`}
    >
      <div className="nabi-action-buttons">
        <IconButton
          color="primary"
          className="nabi-float-right nabi-margin-left-xsmall"
          aria-label="Delete"
          onClick={() => props.deleteEducation(props.id)}
        >
          <Icon>delete</Icon>
        </IconButton>

        <IconButton
          color="primary"
          className="nabi-float-right"
          aria-label="Edit"
          onClick={() => props.editEducation(props.id)}
        >
          <Icon>edit</Icon>
        </IconButton>
      </div>

      <Typography className="nabi-text-uppercase" variant="body2">
        <mark>{props.school}</mark>
      </Typography>

      <Typography className="nabi-margin-top-xsmall">
        <mark>
          {EducationAddedComponent.concentration.replace(
            EducationAddedComponent.degreeTypePlaceholder,
            degreeTypeLabels[props.degreeType]
          ).replace(
            EducationAddedComponent.fieldOfStudyPlaceholder,
            props.fieldOfStudy
          )}
        </mark>
      </Typography>

      <Typography>
        <mark>{props.graduationYear}</mark>
      </Typography>

      <Typography>
        <mark>{props.schoolLocation}</mark>
      </Typography>
    </Grid>
  );
};

export default EducationAdded;
