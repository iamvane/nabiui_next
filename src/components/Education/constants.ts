export namespace EducationComponent {
  export enum Text {
    YourEducation = 'Your Education',
    TellStudentsAbout = 'Tell students about your educational background',
    AddEducation = 'Add Education'
  }
}

export namespace EducationFormComponent {
  export const extraYears = 7;

  export enum Text {
    AddButton = 'Add',
    SaveChangesButton = 'Save Changes',
    CancelButton = 'Cancel'
  }

  export enum FieldNames {
    School = 'school',
    GraduationYear = 'graduationYear',
    FieldOfStudy = 'fieldOfStudy',
    DegreeType = 'degreeType',
    SchoolLocation = 'schoolLocation'
  }

  export enum Ids {
    School = 'school',
    GraduationYear = 'graduationYear',
    FieldOfStudy = 'fieldOfStudy',
    DegreeType = 'degreeType',
    SchoolLocation = 'schoolLocation'
  }

  export enum Placeholders {
    School = 'School Name',
    FieldOfStudy = 'Field of study (major or concentration)',
    SchoolLocation = 'School location (city, state, country)'
  }

  export enum DisabledPlaceholders {
    GraduationYear= 'Graduation year',
    Degreetype = 'Degree type'
  }
}

export const degreeTypeKey = {
  associate: 'associate',
  bachelors: 'bachelors',
  graduate: 'graduate',
  professional: 'professional',
  certification: 'certification',
  other: 'other'
};

export const degreeTypeLabels = {
  [degreeTypeKey.associate]: 'Associate Degree',
  [degreeTypeKey.bachelors]: 'Bachelor\'s Degree',
  [degreeTypeKey.graduate]: 'Graduate Degree',
  [degreeTypeKey.professional]: 'Professional Degree',
  [degreeTypeKey.certification]: 'Certification',
  [degreeTypeKey.other]: 'Other'
};

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degree',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface Degree {
  value: string;
  label: DegreeType;
}

export const degreeOptions = {
  Associate: <Degree> {
    value: degreeTypeKey.associate,
    label: degreeTypeLabels.associate
  },
  Bachelor: <Degree> {
    value: degreeTypeKey.bachelors,
    label: degreeTypeLabels.bachelors
  },
  graduate: <Degree> {
    value: degreeTypeKey.graduate,
    label: degreeTypeLabels.graduate
  },
  Professional: <Degree> {
    value: degreeTypeKey.professional,
    label: degreeTypeLabels.professional
  },
  Certification: <Degree> {
    value: degreeTypeKey.certification,
    label: degreeTypeLabels.certification
  },
  Other: <Degree> {
    value: degreeTypeKey.other,
    label: degreeTypeLabels.other
  }
};

export namespace EducationAddedComponent {
  export const concentration = '{degreeTypeReplace}, {fieldOfStudyReplace}';
  export const degreeTypePlaceholder = '{degreeTypeReplace}';
  export const fieldOfStudyPlaceholder = '{fieldOfStudyReplace}';
}
