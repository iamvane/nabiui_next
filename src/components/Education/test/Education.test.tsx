import * as React from 'react';
import { shallow } from 'enzyme';

import { Education } from 'components/Education/Education';
import { EducationType } from 'components/Education/model';

describe('Education', () => {
  let wrapper: any;
  const mockFunction = jest.fn();

  const mockEducation: EducationType = {
    id: 0,
    school: 'UMass',
    graduationYear: '2012',
    degreeType: 'bachelors',
    fieldOfStudy: 'Music Education',
    schoolLocation: 'Boston MA'
  };

  beforeEach(() => {
    wrapper = shallow(
      <Education
        education={[mockEducation]}
        fetchEducation={mockFunction}
        addEducation={mockFunction}
        editEducation={mockFunction}
        deleteEducation={mockFunction}
        isFetchEducationRequesting={false}
        isAddEducationRequesting={false}
        isDeleteEducationRequesting={false}
        isEditEducationRequesting={false}
        fetchEducationError=""
        addEducationError=""
        editEducationError=""
        deleteEducationError=""
      />
    );
  });

  describe('When the education form is hidden', () => {
    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When the education form is displayed', () => {
    beforeEach(() => {
      wrapper.setState({showEducationForm: true});
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Method handleChange()', () => {
    function test_handleChange(theName: string, theValue: any): void {
      describe(`When the event contains ${theName} for name and ${theValue} for value`, () => {
        beforeEach(() => {
          const e = {
            currentTarget: {
              name: theName,
              value: theValue
            }
          };

          wrapper.instance().handleChange(e);
        });

        it(`Sets the state's ${theName} to ${theValue}`, () => {
          expect(wrapper.state(theName)).toBe(theValue);
        });
      });
    }

    test_handleChange('school', 'UMass Boston');
    test_handleChange('graduationYear', 2012);
  });

  describe('Method resetState()', () => {
    beforeEach(() => {
      wrapper.instance().resetState();
    });

    it('Resets the state', () => {
      expect(wrapper.state('id')).toBe(undefined);
      expect(wrapper.state('school')).toBe('');
      expect(wrapper.state('graduationYear')).toBe('');
      expect(wrapper.state('degreeType')).toBe('');
      expect(wrapper.state('fieldOfStudy')).toBe('');
      expect(wrapper.state('schoolLocation')).toBe('');
    });
  });
});
