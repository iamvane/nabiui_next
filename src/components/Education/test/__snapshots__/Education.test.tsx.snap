// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education When the education form is displayed Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Education
    addEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    addEducationError=""
    deleteEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    deleteEducationError=""
    editEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    editEducationError=""
    education={
      Array [
        Object {
          "degreeType": "bachelors",
          "fieldOfStudy": "Music Education",
          "graduationYear": "2012",
          "id": 0,
          "school": "UMass",
          "schoolLocation": "Boston MA",
        },
      ]
    }
    fetchEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      }
    }
    fetchEducationError=""
    isAddEducationRequesting={false}
    isDeleteEducationRequesting={false}
    isEditEducationRequesting={false}
    isFetchEducationRequesting={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall nabi-text-uppercase"
            variant="body2"
          >
            Add Education
          </WithStyles(Typography)>
          <EducationForm
            allFieldsFilled={false}
            degreeType=""
            fieldOfStudy=""
            graduationYear=""
            handleCancel={[Function]}
            handleChange={[Function]}
            handleOnBlur={[Function]}
            handleSave={[Function]}
            isEditing={false}
            school=""
            schoolLocation=""
          />
        </div>,
        <Unknown
          backPath="job-preferences"
          buttonText="Continue"
          divider={true}
          errors=""
          handleNext={[Function]}
          icon={
            <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>
          }
          nextPath="employment"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall nabi-text-uppercase"
              variant="body2"
            >
              Add Education
            </WithStyles(Typography)>,
            <EducationForm
              allFieldsFilled={false}
              degreeType=""
              fieldOfStudy=""
              graduationYear=""
              handleCancel={[Function]}
              handleChange={[Function]}
              handleOnBlur={[Function]}
              handleSave={[Function]}
              isEditing={false}
              school=""
              schoolLocation=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Education",
              "className": "nabi-margin-top-xsmall nabi-text-uppercase",
              "variant": "body2",
            },
            "ref": null,
            "rendered": "Add Education",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "allFieldsFilled": false,
              "degreeType": "",
              "fieldOfStudy": "",
              "graduationYear": "",
              "handleCancel": [Function],
              "handleChange": [Function],
              "handleOnBlur": [Function],
              "handleSave": [Function],
              "isEditing": false,
              "school": "",
              "schoolLocation": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backPath": "job-preferences",
          "buttonText": "Continue",
          "divider": true,
          "errors": "",
          "handleNext": [Function],
          "icon": <WithStyles(Icon)>
            arrow_forward
          </WithStyles(Icon)>,
          "nextPath": "employment",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall nabi-text-uppercase"
              variant="body2"
            >
              Add Education
            </WithStyles(Typography)>
            <EducationForm
              allFieldsFilled={false}
              degreeType=""
              fieldOfStudy=""
              graduationYear=""
              handleCancel={[Function]}
              handleChange={[Function]}
              handleOnBlur={[Function]}
              handleSave={[Function]}
              isEditing={false}
              school=""
              schoolLocation=""
            />
          </div>,
          <Unknown
            backPath="job-preferences"
            buttonText="Continue"
            divider={true}
            errors=""
            handleNext={[Function]}
            icon={
              <WithStyles(Icon)>
                arrow_forward
              </WithStyles(Icon)>
            }
            nextPath="employment"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className="nabi-margin-top-xsmall nabi-text-uppercase"
                variant="body2"
              >
                Add Education
              </WithStyles(Typography)>,
              <EducationForm
                allFieldsFilled={false}
                degreeType=""
                fieldOfStudy=""
                graduationYear=""
                handleCancel={[Function]}
                handleChange={[Function]}
                handleOnBlur={[Function]}
                handleSave={[Function]}
                isEditing={false}
                school=""
                schoolLocation=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Education",
                "className": "nabi-margin-top-xsmall nabi-text-uppercase",
                "variant": "body2",
              },
              "ref": null,
              "rendered": "Add Education",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "allFieldsFilled": false,
                "degreeType": "",
                "fieldOfStudy": "",
                "graduationYear": "",
                "handleCancel": [Function],
                "handleChange": [Function],
                "handleOnBlur": [Function],
                "handleSave": [Function],
                "isEditing": false,
                "school": "",
                "schoolLocation": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backPath": "job-preferences",
            "buttonText": "Continue",
            "divider": true,
            "errors": "",
            "handleNext": [Function],
            "icon": <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>,
            "nextPath": "employment",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Education When the education form is hidden Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Education
    addEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    addEducationError=""
    deleteEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    deleteEducationError=""
    editEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    editEducationError=""
    education={
      Array [
        Object {
          "degreeType": "bachelors",
          "fieldOfStudy": "Music Education",
          "graduationYear": "2012",
          "id": 0,
          "school": "UMass",
          "schoolLocation": "Boston MA",
        },
      ]
    }
    fetchEducation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchEducationError=""
    isAddEducationRequesting={false}
    isDeleteEducationRequesting={false}
    isEditEducationRequesting={false}
    isFetchEducationRequesting={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <SectionTitle
            text="Your Education"
          />
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall"
          >
            Tell students about your educational background
          </WithStyles(Typography)>
          <ul>
            <li
              className="nabi-list"
            >
              <EducationAdded
                degreeType="bachelors"
                deleteEducation={[Function]}
                editEducation={[Function]}
                fieldOfStudy="Music Education"
                graduationYear="2012"
                gridWidth={6}
                id={0}
                school="UMass"
                schoolLocation="Boston MA"
              />
            </li>
          </ul>
          <div
            className="nabi-margin-top-medium"
          >
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              <WithStyles(Icon)
                className="nabi-margin-right-xsmall"
              >
                add
              </WithStyles(Icon)>
              Add Education
            </WithStyles(Button)>
          </div>
        </div>,
        <Unknown
          backPath="job-preferences"
          buttonText="Continue"
          divider={true}
          errors=""
          handleNext={[Function]}
          icon={
            <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>
          }
          nextPath="employment"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SectionTitle
              text="Your Education"
            />,
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall"
            >
              Tell students about your educational background
            </WithStyles(Typography)>,
            <ul>
              <li
                className="nabi-list"
              >
                <EducationAdded
                  degreeType="bachelors"
                  deleteEducation={[Function]}
                  editEducation={[Function]}
                  fieldOfStudy="Music Education"
                  graduationYear="2012"
                  gridWidth={6}
                  id={0}
                  school="UMass"
                  schoolLocation="Boston MA"
                />
              </li>
            </ul>,
            <div
              className="nabi-margin-top-medium"
            >
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(Icon)
                  className="nabi-margin-right-xsmall"
                >
                  add
                </WithStyles(Icon)>
                Add Education
              </WithStyles(Button)>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "Your Education",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Tell students about your educational background",
              "className": "nabi-margin-top-xsmall",
            },
            "ref": null,
            "rendered": "Tell students about your educational background",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nabi-list"
                >
                  <EducationAdded
                    degreeType="bachelors"
                    deleteEducation={[Function]}
                    editEducation={[Function]}
                    fieldOfStudy="Music Education"
                    graduationYear="2012"
                    gridWidth={6}
                    id={0}
                    school="UMass"
                    schoolLocation="Boston MA"
                  />
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <EducationAdded
                    degreeType="bachelors"
                    deleteEducation={[Function]}
                    editEducation={[Function]}
                    fieldOfStudy="Music Education"
                    graduationYear="2012"
                    gridWidth={6}
                    id={0}
                    school="UMass"
                    schoolLocation="Boston MA"
                  />,
                  "className": "nabi-list",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "degreeType": "bachelors",
                    "deleteEducation": [Function],
                    "editEducation": [Function],
                    "fieldOfStudy": "Music Education",
                    "graduationYear": "2012",
                    "gridWidth": 6,
                    "id": 0,
                    "school": "UMass",
                    "schoolLocation": "Boston MA",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(Icon)
                  className="nabi-margin-right-xsmall"
                >
                  add
                </WithStyles(Icon)>
                Add Education
              </WithStyles(Button)>,
              "className": "nabi-margin-top-medium",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Icon)
                    className="nabi-margin-right-xsmall"
                  >
                    add
                  </WithStyles(Icon)>,
                  "Add Education",
                ],
                "color": "primary",
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "add",
                    "className": "nabi-margin-right-xsmall",
                  },
                  "ref": null,
                  "rendered": "add",
                  "type": [Function],
                },
                "Add Education",
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backPath": "job-preferences",
          "buttonText": "Continue",
          "divider": true,
          "errors": "",
          "handleNext": [Function],
          "icon": <WithStyles(Icon)>
            arrow_forward
          </WithStyles(Icon)>,
          "nextPath": "employment",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <SectionTitle
              text="Your Education"
            />
            <WithStyles(Typography)
              className="nabi-margin-top-xsmall"
            >
              Tell students about your educational background
            </WithStyles(Typography)>
            <ul>
              <li
                className="nabi-list"
              >
                <EducationAdded
                  degreeType="bachelors"
                  deleteEducation={[Function]}
                  editEducation={[Function]}
                  fieldOfStudy="Music Education"
                  graduationYear="2012"
                  gridWidth={6}
                  id={0}
                  school="UMass"
                  schoolLocation="Boston MA"
                />
              </li>
            </ul>
            <div
              className="nabi-margin-top-medium"
            >
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(Icon)
                  className="nabi-margin-right-xsmall"
                >
                  add
                </WithStyles(Icon)>
                Add Education
              </WithStyles(Button)>
            </div>
          </div>,
          <Unknown
            backPath="job-preferences"
            buttonText="Continue"
            divider={true}
            errors=""
            handleNext={[Function]}
            icon={
              <WithStyles(Icon)>
                arrow_forward
              </WithStyles(Icon)>
            }
            nextPath="employment"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SectionTitle
                text="Your Education"
              />,
              <WithStyles(Typography)
                className="nabi-margin-top-xsmall"
              >
                Tell students about your educational background
              </WithStyles(Typography)>,
              <ul>
                <li
                  className="nabi-list"
                >
                  <EducationAdded
                    degreeType="bachelors"
                    deleteEducation={[Function]}
                    editEducation={[Function]}
                    fieldOfStudy="Music Education"
                    graduationYear="2012"
                    gridWidth={6}
                    id={0}
                    school="UMass"
                    schoolLocation="Boston MA"
                  />
                </li>
              </ul>,
              <div
                className="nabi-margin-top-medium"
              >
                <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  variant="contained"
                >
                  <WithStyles(Icon)
                    className="nabi-margin-right-xsmall"
                  >
                    add
                  </WithStyles(Icon)>
                  Add Education
                </WithStyles(Button)>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "Your Education",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Tell students about your educational background",
                "className": "nabi-margin-top-xsmall",
              },
              "ref": null,
              "rendered": "Tell students about your educational background",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="nabi-list"
                  >
                    <EducationAdded
                      degreeType="bachelors"
                      deleteEducation={[Function]}
                      editEducation={[Function]}
                      fieldOfStudy="Music Education"
                      graduationYear="2012"
                      gridWidth={6}
                      id={0}
                      school="UMass"
                      schoolLocation="Boston MA"
                    />
                  </li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <EducationAdded
                      degreeType="bachelors"
                      deleteEducation={[Function]}
                      editEducation={[Function]}
                      fieldOfStudy="Music Education"
                      graduationYear="2012"
                      gridWidth={6}
                      id={0}
                      school="UMass"
                      schoolLocation="Boston MA"
                    />,
                    "className": "nabi-list",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "degreeType": "bachelors",
                      "deleteEducation": [Function],
                      "editEducation": [Function],
                      "fieldOfStudy": "Music Education",
                      "graduationYear": "2012",
                      "gridWidth": 6,
                      "id": 0,
                      "school": "UMass",
                      "schoolLocation": "Boston MA",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  variant="contained"
                >
                  <WithStyles(Icon)
                    className="nabi-margin-right-xsmall"
                  >
                    add
                  </WithStyles(Icon)>
                  Add Education
                </WithStyles(Button)>,
                "className": "nabi-margin-top-medium",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(Icon)
                      className="nabi-margin-right-xsmall"
                    >
                      add
                    </WithStyles(Icon)>,
                    "Add Education",
                  ],
                  "color": "primary",
                  "onClick": [Function],
                  "variant": "contained",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "add",
                      "className": "nabi-margin-right-xsmall",
                    },
                    "ref": null,
                    "rendered": "add",
                    "type": [Function],
                  },
                  "Add Education",
                ],
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backPath": "job-preferences",
            "buttonText": "Continue",
            "divider": true,
            "errors": "",
            "handleNext": [Function],
            "icon": <WithStyles(Icon)>
              arrow_forward
            </WithStyles(Icon)>,
            "nextPath": "employment",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
